index.js

```diff
@@ -12,34 +14,17 @@ const mkdirp = require('mkdirp')
 const rimraf = require('rimraf')
 const path = require('path')
 const signalExit = require('signal-exit')
-const home = process.env.SPAWN_WRAP_SHIM_ROOT || require('os-homedir')()
-const homedir = home + '/.node-spawn-wrap-'
-const which = require('which')
-const util = require('util')
-
-const doDebug = process.env.SPAWN_WRAP_DEBUG === '1'
-
-function debug () {
-  if (!doDebug) {
-    return
-  }
-  const prefix = 'SW ' + process.pid + ': '
-  const data = util.format.apply(util, arguments).trim()
-  const message = prefix + data.split('\n').join('\n' + prefix)
-  process.stderr.write(message + '\n')
-}
+const {IS_DEBUG, debug} = require("./lib/debug")
+const {munge} = require("./lib/munge")
+const {homedir} = require("./lib/homedir")
```

Comment:
I think it would be best to export these as defaults instead of objects from the modules - it matches the philosophy/intent of this PR, rather than leaving it open to add extra exports to those small focused modules.

Reply:
I prefer to leave it this way for the moment because it's easier to add extra functions without having to chose a default one. We could revisit it afterwards?

<sub>I also hope that using named exports will make it easier to migrate to ESM when Node supports it...</sub>

Reply:
I can understand what @JaKXz is suggesting, you've split things so each source intentionally has a single function.  Using the default export would discourage going against that.

TBH I don't understand how ESM would effect this module directly, we can easily use default exports if this module ever migrates to ESM.

Reply:
Regarding ESM, I feel that it is more natural to use named exports. For example it plays better with the IDE's autocomplete/autoimport features. There's also some uncertainty around how bundlers handle default exports compared to the discussion around Node's native ESM.

But you're right, it's better to worry about the future later: it's weird to use such exports for single-export commonjs modules.

Reply:
Fixed